### PR TITLE
Allows uniforms to have multiple accessories attached, removes armor from accessories

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

image

- Multiple accessories can now be pinned to one jumpsuit. 
    - All accessories are shown on examine, but only the first is shown on-mob. 

- Under-clothing code cleanup.

- Removed armor from all accessories. 

- Halved the mood given from Hope to accommodate the fact that it can now stack with others. 

- Generalizes dogtags. Now dogtags will show their string on examine_more if examining the mob wearing it, so you can see the contents of someone's Allergy Dogtag without removing their clothes. 

## Why It's Good For The Game

Accessories are a surprisingly contentious slot for something supposed to be entirely cosmetic, so I've decided to address that.

This turns accessories into almost entirely cosmetic, with some minor effects like mood boosts or pen storage.

People will no longer have to choose between showing off a pride pen or getting a medal from the HoP, or between wearing sick lavaland drip items, while also letting us stretch our legs a bit without worrying about clothing balance. 

## Changelog

:cl: Melbert
add: You can now wear multiple accessories on your uniform at once (up to five by default)
balance: Removed armor from accessories, and nerfs the effects of some accessories. 
/:cl:

### CODE REVIEW

**Comment by CoiledLamb** [comment]
i have been waiting for this PR for months, so thank you!!! the loss of armor and mood is nothing compared to how happy i am to not have to remove my pride pin so the capt can pin medals on me.

**Comment by Enbyy** [comment]
Based MrMelbert pr

**Comment by Pickle-Coding** [comment]
Didn't even know accessories had armour. That's kinda silly.

**Comment by SinfulBliss** [comment]
This removes the purpose of the skull codpiece and bone talisman, which were little mini armor boosts miners could get for slaying some basic mobs.

I don’t see why you couldn’t just make it so the armored accessories wouldn’t be stackable. Why does every fix have to come with a removal of a fun feature (like fry oil) :/

File: code/modules/clothing/suits/_suits.dm [hunk 1]
```diff
@@ -30,14 +30,16 @@
...
-	var/obj/item/clothing/under/U = M.w_uniform
-	if(istype(U) && U.attached_accessory)
-		var/obj/item/clothing/accessory/A = U.attached_accessory
-		if(A.above_suit)
-			. += U.accessory_overlay
+	var/obj/item/clothing/under/undershirt = wearer.w_uniform
+	if(!istype(undershirt) || !LAZYLEN(undershirt.attached_accessories))
+		return
+
+	var/obj/item/clothing/accessory/displayed = undershirt.attached_accessories[1]
```
Comment (ChungusGamer666) [inline code/modules/clothing/suits/_suits.dm:40]: why not loop through the list and display all of the above_suit accessories?

**Comment by ChungusGamer666** [comment]
> This removes the purpose of the skull codpiece and bone talisman, which were little mini armor boosts miners could get for slaying some basic mobs.
> 
> I don’t see why you couldn’t just make it so the armored accessories wouldn’t be stackable. Why does every fix have to come with a removal of a fun feature (like fry oil) :/

I have to agree, this is silly. Surely you could have come up with a way to ensure armor boosting couldn't be stacked instead?

File: code/modules/clothing/suits/_suits.dm (hunk 1 above)
Comment (MrMelbert) [reply code/modules/clothing/suits/_suits.dm:40]: It'd look horrible, they're all overlayed onto the same spot

**Comment by MrMelbert** [comment]
> I don’t see why you couldn’t just make it so the armored accessories wouldn’t be stackable. Why does every fix have to come with a removal of a fun feature (like fry oil) :/

NGL I really struggle to see how adding a necklace that gives you 5% armor to all types then you completely ignore for the rest of the round is a "fun feature", if anything it's more busywork for miners (gotta get my powergame 5% armor)

**Comment by YehnBeep** [comment]
> > I don’t see why you couldn’t just make it so the armored accessories wouldn’t be stackable. Why does every fix have to come with a removal of a fun feature (like fry oil) :/
> 
> NGL I really struggle to see how adding a necklace that gives you 5% armor to all types then you completely ignore for the rest of the round is a "fun feature", if anything it's more busywork for miners (gotta get my powergame 5% armor)

Same as any progression mechanic. People that are into it like having things to work towards, and miner, by its nature of gameplay being built around progression, is going to attract people that like that sort of thing.

**Comment by necromanceranne** [comment]
Fellas, not to burst your bubble, but you're arguing over a 5% difference in armor that ultimately won't even contribute more than 2% to any calculation due to armor multiplication from layer stacking.

**Comment by ChungusGamer666** [comment]
> Fellas, not to burst your bubble, but you're arguing over a 5% difference in armor that ultimately won't even contribute more than 2% to any calculation due to armor multiplication from layer stacking. Who gives a fuck. It's practically a placebo at this point.

"We cucked armor stacking against your will therefore any armor that isn't in the suit slot might as well be removed, get fucked!"

**Comment by Pickle-Coding** [comment]
Allowing accessories to have armour seems like it will just add extra complexity to the system to keep it balanced. Minor armour on accessories that are barely visible to the player just doesn't seem that interesting or worth adding the extra maintenance and complexity for something designed to be cosmetic.

It's pretty much 5% protection at most, you're not really losing anything meaningful.

**Comment by rasepretep** [comment]
Why not make it so that only the first item added to a suit actually gives its specific armor bonus, and anything added after that is ignored. As if the suit had 1 active slot and several passive/cosmetic slots.

**Comment by SpaceVampire** [comment]
Why do you have to nerf the mood on Hope, if there isn't any benefit to wearing it with other accessories anyway? Are you worried about people being able to stack it? Do moods work like that? It just seems unnecessary, and you normally only have one of them anyway.

**Comment by Pickle-Coding** [comment]
> Why do you have to nerf the mood on Hope, if there isn't any benefit to wearing it with other accessories anyway? Are you worried about people being able to stack it? Do moods work like that? It just seems unnecessary, and you normally only have one of them anyway.

Probably because being able to passively ignore most bad mood events, and pretty much nullify any (except depression) mood events from quirks doesn't provide interesting gameplay. How can an emblem make you not care at all about getting your tail ripped out?

**Comment by RedBaronFlyer** [comment]
I thought this was the PR to add cosmetic layouts like Mothblocks alluded to earlier this year, bummer.

**Comment by ChungusGamer666** [comment]
> 

Repeat it with me.
**It's a god damn reward for killing megafauna. It's okay to give it buffs.**

**Comment by MrMelbert** [comment]
> Why do you have to nerf the mood on Hope, if there isn't any benefit to wearing it with other accessories anyway? Are you worried about people being able to stack it? Do moods work like that? It just seems unnecessary, and you normally only have one of them anyway.

Like I said in the PR, to accomodate the fact that you can wear it with other mood accessories like mime/clown fan

> Why not make it so that only the first item added to a suit actually gives its specific armor bonus, and anything added after that is ignored. As if the suit had 1 active slot and several passive/cosmetic slots.

It's possible but juggling passives / actives would greatly complicate the code and is not immediately intuitive

**Comment by necromanceranne** [comment]
> "We cucked armor stacking against your will therefore any armor that isn't in the suit slot might as well be removed, get fucked!"

You are choosing to contribute to the codebase, bob, nobody is forcing anything on you. Resolve it on the downstream you keep advertising if you care about it. It isn't hard to revert. Nor does it change the reality of our codebase and how little impact armor accessories make, so my point stands. 

I much prefer the opportunity for more accessories (and hopefully a way to resolve the layering issues later) than worry about a minor armor boost being lost.

(Alternatively someone could do something more interesting with those accessories if there are worries about ashwalkers being screwed over)

**Comment by DaydreamIQ** [comment]
Iirc the benefit of the talisman/codpiece that people like isn't the 5% armor boost. Its the significant explosive resistance it gives you to avoid getting your limbs blown off by gibtonite. If you're gonna nerf the accessories could you at least move that explosive resist to the gladiator and explorer suits?

**Comment by Pickle-Coding** [comment]
> Iirc the benefit of the talisman/codpiece that people like isn't the 5% armor boost. Its the significant explosive resistance it gives you to avoid getting your limbs blown off by gibtonite. If you're gonna nerf the accessories could you at least move that explosive resist to the gladiator and explorer suits?

Just run away instead of getting hugboxed by your God armour.

**Comment by rasepretep** [comment]
> I don't think killing a simplemob with big sprite and choreographed dancing moves 10 minutes in every round is a very difficult achievement. Also a passive mood buff for the entire round is just simply not an interesting or fun reward. It just makes you be able to ignore a mechanic for most of the round. I don't care how hard it would be to get it, it's just simply bad design.

Are you confusing Pandora with Hierophant? Because I don't recall ever seeing more than 1 Pulsating Tumor spawn, there is a 1:4 chance that the mob spawned from it will be Pandora, and it has to be made sentient to give a 50% chance to drop Hope.

That doesn't feel like "10 minutes in every round" kind of deal.

**Comment by MrMelbert** [comment]
> Iirc the benefit of the talisman/codpiece that people like isn't the 5% armor boost. Its the significant explosive resistance it gives you to avoid getting your limbs blown off by gibtonite.

I think a far more interesting solution to this would be allowing Miners to upgrade their explorer gear with a variety of monster parts rather than just goliath plating and each confers a certain armor bonus, IE goliath plates give melee armor, bones give bomb armor, etc.

**Comment by JohnFulpWillard** [comment]
The mining attachment to your suit is monster parts and one of the few good uses of the bones and sinew. It's sad to see it go but I don't think it matters much in the end, it was just a nice thing to have.

**Comment by Namwty** [comment]
It wouldn't be a TG PR if it didn't remove something from the game.

Sounds neat, but don't see why is it needed to remove armor from accessories, while yes it's laughable armor% to miners, who already have  drakes armor and what not. It is still nice for early experience for Ghost roles and Assistants or generally anyone who just don't have mining gear round start. And yes that talisman can mean a lot when you have no mining gear.

Just make it so it has one armored accessory limit, again I don't think it's big loss, but it's just silly to just remove stuff, especially if it's arguably good for some inexperienced folk.

**Comment by MrMelbert** [comment]
> It wouldn't be a TG PR if it didn't remove something from the game.

It's possible to share an opinion without an insult

**Comment by Cadunkus** [comment]
Good change except for shafting the miner accessories. Just make the armored accessories unstackable.

**Comment by Redrover1760** [comment]
> > Iirc the benefit of the talisman/codpiece that people like isn't the 5% armor boost. Its the significant explosive resistance it gives you to avoid getting your limbs blown off by gibtonite. If you're gonna nerf the accessories could you at least move that explosive resist to the gladiator and explorer suits?
> 
> Just run away instead of getting hugboxed by your God armour.
> 
> > Repeat it with me.
> > It's a god damn reward for killing megafauna. It's okay to give it buffs.
> 
> I don't think killing a simplemob with big sprite and choreographed dancing moves 10 minutes in every round is a very difficult achievement. Also a passive mood buff for the entire round is just simply not an interesting or fun reward. It just makes you be able to ignore a mechanic for most of the round. I don't care how hard it would be to get it, it's just simply bad design.
> 
> I don't want to fill up the comments too much on someone else's PR, so I won't respond any further.

This is mistaken. Hope is a chance reward for killing sentient Pandoras, an elite mob.

**Comment by Pickle-Coding** [comment]
This PR got merged 4 months ago and the information you provided is unimportant. Shoo.

**Pull request merged**